sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/ScriptTransformation.scala

```diff
@@ -53,28 +78,176 @@ case class ScriptTransformation(
       val inputStream = proc.getInputStream
       val outputStream = proc.getOutputStream
       val reader = new BufferedReader(new InputStreamReader(inputStream))
+ 
+      val outputSerde: AbstractSerDe = if (ioschema.outputSerdeClass != "") {
```

Comment:
Make the input / output SerDe in another function(s)? I am not sure if there any code can be shared, but seems some duplicated code.